File: src/frontend/src/app/types/todo.ts
export interface TodoItem {
  id: string
  familyId: string
  title: string
  description: string | null
  isDone: boolean
  assignedToUserId: string | null
  dueDateUtc: string | null
  createdAtUtc: string
  completedAtUtc: string | null
}

export interface CreateTodoRequest {
  title: string
  description?: string | null
  dueDateUtc?: string | null
  assignedToUserId?: string | null
}

File: src/ios/FamilyHub/Models/Todo.swift
import Foundation

struct TodoItem: Codable, Identifiable, Equatable {
    let id: UUID
    let familyId: UUID
    let title: String
    let description: String?
    let isDone: Bool
    let assignedToUserId: UUID?
    let dueDateUtc: Date?
    let createdAtUtc: Date
    let completedAtUtc: Date?

    var isOverdue: Bool {
        guard let due = dueDateUtc, !isDone else { return false }
        return due < Date()
    }
}

struct CreateTodoRequest: Encodable {
    let title: String
    let description: String?
    let dueDateUtc: String?
    let assignedToUserId: String?
}
